Mathlib/Algebra/Ring/Subsemiring/Defs.lean

```diff
@@ -104,12 +104,7 @@ instance (priority := 75) toSemiring {R} [Semiring R] [SetLike S R] [Subsemiring
   Subtype.coe_injective.semiring Subtype.val rfl rfl (fun _ _ => rfl) (fun _ _ => rfl)
     (fun _ _ => rfl) (fun _ _ => rfl) fun _ => rfl
 
-@[simp, norm_cast]
-theorem coe_pow {R} [Monoid R] [SetLike S R] [SubmonoidClass S R] (x : s) (n : ℕ) :
-    ((x ^ n : s) : R) = (x : R) ^ n := by
-  induction n with
-  | zero => simp
-  | succ n ih => simp [pow_succ, ih]
+@[deprecated (since := "2025-07-29")] alias coe_pow := SubmonoidClass.coe_pow
```

Comment:
Should this be an alias or a deprecation in favour of `SubmonoidClass.coe_pow`?

Reply:
I'm not sure I follow your question - my proposal is to deprecate the redundant `SubsemiringClass.coe_pow`.

Reply:
I don't follow my question any more either --- let's call this good enough.